src/hotspot/share/prims/jvmtiThreadState.cpp

```diff
@@ -554,13 +554,14 @@ JvmtiVTMSTransitionDisabler::VTMS_vthread_end(jobject vthread) {
       JvmtiExport::post_vthread_end(vthread);
     }
   }
+  VTMS_unmount_begin(vthread, /* last_unmount */ true);
   if (thread->jvmti_thread_state() != nullptr) {
+    assert(thread->jvmti_thread_state()->is_virtual(), "wrong JvmtiThreadState");
```

Comment:
We agreed with you to temporarily remove this assert as it triggers the bug:
[8308124](https://bugs.openjdk.org/browse/JDK-8308124) dynamic loading of a JVMTI agent has a race with JvmtiThreadState cleanup

A fix of the [8308124](https://bugs.openjdk.org/browse/JDK-8308124) will add this assert back.

Reply:
Yes, thanks for finding and filing a bug for that. I removed the assert.